### PR TITLE
Seller Experience - Add new design picker UI 

### LINKED ISSUE #64715: Seller Experience - Build new design picker UI
Following on from https://github.com/Automattic/wp-calypso/issues/64706 as part of the new Seller Experience theme picker designs described in paYKcK-1Hs-p2, we need to update the UI for the design selection screen:

image

Requirements: 
* General design tweaks
* Restore the category picker on the screen's left side
* When the user selects a particular theme, they are no longer prompted to select the theme OR preview it; instead, they will always click on the theme and preview it first.

Figma link:  0EogtxvlmxMGcnEEaMVECv-fi

### PR DESCRIPTION
**Pull request by valterlorran**
#### Proposed Changes

* Changes the design picker UI 

#### Testing Instructions

* Go to `/setup/designSetup?siteSlug=[SITE-SLUG]&flags=signup/theme-preview-screen`
* On the design picker step, categories should be visible
* Clicking on a theme, you should be redirected to its preview page (the same should happen in other flows)

Closes #64715

-----

After the `signup/theme-preview-screen` feature flag is enabled everywhere, we need to remove 
```
/* Hide categories from the sell intent */
.design-picker__sell-intent {
	.design-picker-category-filter__sidebar,
	.design-picker-category-filter__dropdown,
	.design-picker__category-heading-0 {
		display: none;
	}
}
```
from `client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/style.scss`

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/new-site-setup-design-picker.tsx
Comment (sixhours) [inline client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/new-site-setup-design-picker.tsx]: Creating a new component works, but for clarity we probably want to name it something more descriptive -- I have visions of some kind of "NewSetupDesignPickerV2Final" situation in the future. :D 

Maybe `SellerDesignPicker`?

**Comment by sixhours** [comment]
> Depends on https://github.com/Automattic/wp-calypso/pull/64821

It may be easier to test this while #64821 is waiting for review by setting the base branch for this PR to `add/hide-design-picker-buttons-option` instead of `trunk`:

Screen Shot 2022-06-21 at 3 30 56 PM 1

You can change it by editing the PR title.

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/new-site-setup-design-picker.tsx
Comment (valterlorran) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/new-site-setup-design-picker.tsx]: Great suggestion 😃 I always struggle with naming.

Comment (blackjackkent) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/new-site-setup-design-picker.tsx]: Worth noting that the changes we're making will be applying to all flows and are not seller specific even though we are working on them as part of our Seller Experience initiative. 

I might honestly suggest keeping the existing `SiteSetupDesignPicker` component and having the check happen within its render function, i.e.

```
const SiteSetupDesignPicker: Step = ( { navigation, flow } ) => {
    if ( isEnabled( 'signup/theme-preview-screen' ) ) {
        // ... new logic or split it into its own render function within this file
    } else {
        // ... old logic or split it into its own render function within this file
    }
}
```

That way we can maintain the existing component name which is still accurate.

Thoughts?

**Review by sixhours — APPROVED** [review]
This is working well for me!

**Pull request merged**
